pandas/tests/indexes/multi/test_set_ops.py

```diff
@@ -189,14 +189,62 @@ def test_difference(idx, sort):
         first.difference([1, 2, 3, 4, 5], sort=sort)
 
 
-@pytest.mark.parametrize("sort", [True, False])
+def test_difference_sort_special():
+    # GH-24959
+    idx = pd.MultiIndex.from_product([[1, 0], ['a', 'b']])
+    # sort=None, the default
+    result = idx.difference([])
+    tm.assert_index_equal(result, idx)
+
+
+@pytest.mark.xfail(reason="Not implemented.")
+def test_difference_sort_special_true():
+    # TODO decide on True behaviour
```

Comment:
@TomAugspurger thoughts on what it would take to reach a conclusion about this?  worth adding to the agenda for next week?  I think without a specific effort this is liable to stay failing indefinitely

Reply:
Perhaps I misunderstand, but should the expected result be the current result with `.sort_values()` called on it?